### DISCUSSION TITLE
Render SVG text so each copy of the same glyph is identical

### DISCUSSION THREAD

**Discussion by mincerafter42** [ImageMagick/ImageMagick#6536] [Help]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Linux Mint MATE 21.1

### Description

Is it possible to make each copy of a glyph identical when converting an SVG to a raster format?

e.g. running `echo '<svg xmlns="http://www.w3.org/2000/svg" width="15" height="15"><text y="11">aa</text></svg>' | convert - svg.png` makes a PNG where the two "a"s don't have identical pixels. I want to make them have identical pixels, like generated with `convert label:aa label.png`.  
(The real SVGs, of course, are more complex, such that they can't be manually converted to a `label:`-using command so easily)

**Comment by snibgo** [comment]
Please show your output image. For me, the pixels in the two letters are identical.
```
$ echo '<svg xmlns="http://www.w3.org/2000/svg" width="15" height="15"><text y="11">aa</text></svg>' | magick - +write aa.png -trim -crop 2x1@ -metric RMSE -format %[distortion] -compare info:
0
```
aa.png is:
![aa](https://github.com/ImageMagick/ImageMagick/assets/33812368/85d38480-e689-4ddb-9f94-a8a7bcaeb4d5)

However, the similarity probably depends on the font and the spacing. Remember that most fonts are defined as vectors (not rasters), and if the position of a vector shifts by less than an exact number of pixels, the pixels will be different.

**Reply by mincerafter42** [reply]
Output of `echo '<svg xmlns="http://www.w3.org/2000/svg" width="15" height="15"><text y="11">aa</text></svg>' | convert - svg.png`:
![aa](https://github.com/ImageMagick/ImageMagick/assets/27905767/e01f36e5-05a3-43bd-8655-a430a367d7b0)

**Comment by snibgo** [comment]
Yes, the problem is clear in your example image.

I suggest you experiment with different fonts and text sizes. Fixed-width fonts such as Courier are probably better for you -- if you find a text size that gives text width and height an exact number of pixels, that should work for all glyphs.

**Comment by mincerafter42** [comment]
Figured out the solution: use the prefix `msvg:` for the SVG image.
For the example I gave, it'd be `echo '<svg xmlns="http://www.w3.org/2000/svg" width="15" height="15"><text y="11">aa</text></svg>' | convert msvg:- svg.png`
